backend/graph/api/schema.resolvers.go

```diff
@@ -182,6 +186,59 @@ func (r *queryRootResolver) Export(ctx context.Context, groups []string) (*model
 	}, nil
 }
 
+// Redirect is the resolver for the redirect field.
+func (r *queryRootResolver) Redirect(ctx context.Context, id string) (*model.RedirectLink, error) {
+	ginCtx, err := utils.GinCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	profile := user.GetProfileFromCtx(ginCtx)
+	if profile == nil {
+		return nil, merry.New(
+			"Not authorized",
+			merry.WithUserMessage("you are not authorized for this query"),
+		)
+	}
+
+	redir, err := r.Queries.GetRedirectByCode(ctx, id)
```

Comment:
Do we want a cache here perhaps?

Put it through a dataloader and it should be cached, invalid ids as well.

Reply:
Yes, very good point. This is actually an action that is likely very spiky.